src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasService.kt

```diff
@@ -22,47 +22,41 @@ import io.grpc.stub.StreamObserver
  */
 class CosmasService : CosmasGrpc.CosmasImplBase() {
 
-    private val files: HashMap<String, HashMap<String, ArrayList<ByteString>>> = HashMap()
-    
+    private val files = mutableMapOf<String, MutableList<ByteString>>()
+
     override fun getVersion(request: CosmasProto.GetVersionRequest,
                             responseObserver: StreamObserver<CosmasProto.GetVersionResponse>) {
         println("Get request for version: ${request.version}")
-        val response: CosmasProto.GetVersionResponse
-        response = if (!checkRequest(request)) {
-            println("Bad request")
-            CosmasProto.GetVersionResponse.newBuilder().build()
+        val response = CosmasProto.GetVersionResponse.newBuilder()
+        val fileVersions = files[request.fileId]
+        if (fileVersions != null &&
+                request.version < fileVersions.size &&
+                request.version >= 0) {
+            val file = fileVersions[request.version]
+            response.file = file
         } else {
-            val file = files[request.projectId]!![request.fileId]!![request.version]
-            CosmasProto.GetVersionResponse.newBuilder().setFile(file).build()
+            println("Bad request")

```

Comment:
Good practice is to provide as much useful information as possible in the logged message.  When you see "bad request" in the production logs you need to know: why is this request bad? Because version is negative? Or maybe because it is too big? Anything else?

Reply:
done